### PR TITLE
Fix build errors and warnings for IAR compiler

### PR DESCRIPTION
**Pull request by tony-josi-aws**
Description
-----------
This PR fixes the build errors and compilation warnings for the IAR compiler. 

Test Steps
-----------
Build succeeds without errors or warnings other than unreferenced static functions.

Checklist:
----------
- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------
Build issues and warnings in IAR build.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/FreeRTOS_DHCP.c [hunk 1]
```diff
@@ -100,7 +100,7 @@
 /*
  * Generate a DHCP request packet, and send it on the DHCP socket.
  */
-    static BaseType_t prvSendDHCPRequest( const NetworkEndPoint_t * pxEndPoint );
+    static BaseType_t prvSendDHCPRequest( NetworkEndPoint_t * pxEndPoint );
```
Comment (moninom1) [inline source/FreeRTOS_DHCP.c:103]: Why are we removing const? They might have been added for MISRA fixes

File: source/FreeRTOS_IP_Utils.c [hunk 2]
```diff
@@ -829,10 +829,8 @@ void vPreCheckConfigs( void )
             if( uxSize == 8U )
             {
                 /* This is a 64-bit platform, make sure there is enough space in
-                 * pucEthernetBuffer to store a pointer. */
-                configASSERT( ipconfigBUFFER_PADDING >= 14 );
-                /* But it must have this strange alignment: */
-                configASSERT( ( ( ( ipconfigBUFFER_PADDING ) + 2 ) % 4 ) == 0 );
+                 * pucEthernetBuffer to store a pointer and it must have this strange alignment. */
+                configASSERT( ( ipconfigBUFFER_PADDING >= 14 ) && ( ( ( ( ipconfigBUFFER_PADDING ) + 2 ) % 4 ) == 0 ) );
```
Comment (moninom1) [inline source/FreeRTOS_IP_Utils.c:833]: It should be ||

File: source/FreeRTOS_TCP_Transmission_IPV4.c [hunk 3]
```diff
@@ -349,6 +349,7 @@
         BaseType_t xReturn = pdTRUE;
         uint32_t ulInitialSequenceNumber = 0;
         NetworkEndPoint_t * pxEndPoint = pxSocket->pxEndPoint;
+        ( void ) pxEndPoint;
```
Comment (moninom1) [inline source/FreeRTOS_TCP_Transmission_IPV4.c:352]: This is also a MISRA fix

File: source/FreeRTOS_DHCP.c (hunk 1 above)
Comment (tony-josi-aws) [reply source/FreeRTOS_DHCP.c:103]: Thats becuase in the same function we are assigning it to a non const pointer (`xDHCPv4Socket->pxEndPoint = pxEndPoint;`) and IAR considers this as an error. I think MISRA shouldn't be complaining about that in this case.

File: source/FreeRTOS_IP_Utils.c (hunk 2 above)
Comment (tony-josi-aws) [reply source/FreeRTOS_IP_Utils.c:833]: We need to assert if either `( ipconfigBUFFER_PADDING >= 14 )` or `( ( ( ( ipconfigBUFFER_PADDING ) + 2 ) % 4 ) == 0 )` is false, as `configASSERT` will be triggered if the output of the conditional expression is 0 I think we might need `&&`.

Comment (moninom1) [reply source/FreeRTOS_IP_Utils.c:833]: Yes if one condition is 0 and other is 1, we should ASSERT, so && make sense.

**Pull request closed without merging**